src/Parser/Shaman/Restoration/Modules/Talents/CrashingWaves.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+
+import SpellLink from 'common/SpellLink';
+import SPELLS from 'common/SPELLS';
+import { formatPercentage } from 'common/format';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import HIT_TYPES from 'Parser/Core/HIT_TYPES';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import StatTracker from 'Parser/Core/Modules/StatTracker';
+import calculateEffectiveHealing from 'Parser/Core/calculateEffectiveHealing';
+
+const BUFFER_WAVE = 750;
+const BUFFER_SURGE = 100;
```

Comment:
Why two different buffers?

Reply:
Surge cares about the buff on cast_finish while Wave cares about it at cast_start, I don't know an easy way to check if a buff was active at the start of a cast so i check for a minimum GCD to cover most situations.